projects/cdk/services/theme-color.service.ts

```diff
@@ -0,0 +1,42 @@
+import {DOCUMENT} from '@angular/common';
+import {inject, Injectable} from '@angular/core';
+import {tuiCreateToken} from '@taiga-ui/cdk/utils';
+
+export const TUI_THEME_COLOR = tuiCreateToken<string | null>(null);
+
+@Injectable({
+    providedIn: 'root',
+})
+export class TuiThemeColorService {
+    private readonly doc = inject(DOCUMENT);
+    private readonly initial = inject(TUI_THEME_COLOR);
+
+    constructor() {
+        if (this.initial) {
+            this.setColor(this.initial);
+        }
+    }
+
+    public getColor(): string {
```

Comment:
We don't need `getColor` to query DOM. We can just store `color` as property, instead of a readonly `initial`. Maybe also using setter instead of method to set it.

Reply:
fixed